lib/index.js

```diff
@@ -0,0 +1,261 @@
+'use strict';
+
+const path = require('path');
+const querystring = require('querystring');
+const fetch = require('node-fetch');
```

Comment:
I'm also not thrilled about baking the actual API fetch so hard into the behaviour as it couples us to one provider. I'd probably want to be able to pass a custom model into the behaviour that allows me to modify the fetching behaviour.